src/JMS/Serializer/Metadata/ClassMetadata.php

```diff
@@ -34,6 +34,7 @@ class ClassMetadata extends MergeableClassMetadata
     const ACCESSOR_ORDER_UNDEFINED = 'undefined';
     const ACCESSOR_ORDER_ALPHABETICAL = 'alphabetical';
     const ACCESSOR_ORDER_CUSTOM = 'custom';
+    const COUNT_PARAMS = 16;
```

Comment:
why do not use this approach https://github.com/schmittjoh/serializer/pull/301/files#diff-2d7dc6b1d453d596671e68f15d9a36d0R158 to have a backward compatibile serializtion ? and does not require all che "count" checks